File: SortList/SortList.cpp
#include "../crio/cpp/io/FastIO.hpp"
#include "../crio/cpp/io/PrintMatrix.hpp"
#include "../crio/cpp/io/ReadMatrix.hpp"
#include <bits/stdc++.h>
#include "Solution.cpp"

using namespace std;


int main() {
    FastIO();
    int n;
    cin >> n;
    vector<int> nums;
    ReadMatrix<int>().OneDMatrix(n, nums);
    ListNode* list = ListNode().createList(nums);
    SortList obj;
    ListNode* sortedList = obj.sortList(list);
    vector<int> result = ListNode().extractList(sortedList);
    PrintMatrix<int>().OneDMatrix(result, " ");
    return 0;
}

File: SortList/SortList.java
import java.io.*;
import java.util.*;
import crio.ds.List.ListNode;


class SortList {

    public static void main(String args[]) {

        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();

        List<Integer> nums = new ArrayList<Integer>();

        for (int i = 0 ; i < n ;i++) {
            nums.add(scanner.nextInt());
        }

        ListNode list = ListNode.createList(nums);
        ListNode sortedList = new Solution().sortList(list);
        List<Integer> res = ListNode.extractList(sortedList);
        for (int i = 0; i < res.size(); ++i) {
            System.out.printf("%d ", res.get(i));
        }
    }
}
